tests/modeling/test_inference.py

```diff
@@ -115,3 +115,91 @@ def test_select_top_features_for_display(
     )
     assert isinstance(obs, pd.DataFrame) and not obs.empty
     assert pd.testing.assert_frame_equal(obs, exp) is None
+
+
+@pytest.fixture
+def sample_data():
+    data = {
+        'student_id': [1, 2, 3],
+        'feature1': [0.1, 0.2, 0.3],
+        'feature2': [0.4, 0.5, 0.6]
+    }
+    return pd.DataFrame(data)
+
+# Create dummy KernelExplainer 
+class SimpleKernelExplainer:
+    def shap_values(self, X):
+        # Simulate SHAP values: For simplicity, we return random numbers
+        return np.random.rand(len(X), len(X.columns)) * 0.1  # Random SHAP values between 0 and 0.1
+
+@pytest.fixture
+def explainer():
+    return SimpleKernelExplainer()
+
+@pytest.mark.parametrize(
+    "input_data, expected_shape",
+    [
+        ({"student_id": [1, 2, 3], "feature1": [0.1, 0.2, 0.3], "feature2": [0.4, 0.5, 0.6]}, (3, 3)),
+        ({"student_id": [1, 2], "feature1": [0.1, 0.2], "feature2": [0.4, 0.5]}, (2, 3))
+    ]
+)
+def test_calculate_shap_values_basic(input_data, expected_shape, explainer):
+    df = pd.DataFrame(input_data)
+    student_id_col = 'student_id'
+    model_features = ['feature1', 'feature2']
+    mode = df.mode().iloc[0] 
+    
+    iterator = iter([df])
+    
+    result = list(calculate_shap_values(iterator, student_id_col=student_id_col, model_features=model_features, explainer=explainer, mode=mode))
+    
+    # Check that the result contains the expected number of rows and columns
+    shap_df = result[0]
+    assert shap_df.shape == expected_shape
+    
+    # Ensure that 'student_id' column is present
+    assert student_id_col in shap_df.columns
+    
+    # Ensure that SHAP values are generated and are numeric
+    assert isinstance(shap_df[model_features].iloc[0, 0], (float, np.float64))
+    assert isinstance(shap_df[model_features].iloc[0, 1], (float, np.float64))
```

Comment:
There may be handy pandas functionality for this -- maybe `from pandas.api.types import is_numeric_dtype`